cpp/daal/src/algorithms/pca/transform/oneapi/pca_transform_dense_default_batch_oneapi_impl.i

```diff
@@ -253,8 +263,11 @@ services::Status TransformKernelOneAPI<algorithmFPType, method>::initBuffers(Exe
     DAAL_CHECK_STATUS(status, allocateBuffer(ctx, invSigmas, numFeatures));
     DAAL_CHECK_STATUS(status, allocateBuffer(ctx, invEigenvalues, numComponents));
     DAAL_CHECK_STATUS(status, allocateBuffer(ctx, rawMeans, numFeatures));
+    DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(uint32_t, numVectors, numFeatures);
     copyBlock = ctx.allocate(TypeIds::id<algorithmFPType>(), numVectors * numFeatures, status);
     DAAL_CHECK_STATUS_VAR(status);
+
+    /// check buffer size. which macros should I use?
```

Comment:
You already have checks inside the function.

Reply:
it was a question to u=)